sdk/core/azure-core/azure/core/_stream.py

```diff
@@ -0,0 +1,41 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+from typing import Iterator, AsyncIterator
+from typing_extensions import Protocol, runtime_checkable, Self
+
+
+@runtime_checkable
+class Streamable(Protocol):
```

Comment:
We should use `typing.ContextManager` and `typing.Iterable` as base types? I would also consider making it generic so we can change what we iterator over from `bytes`.

Reply:
Good suggestions!